draft-ietf-ppm-dap.md

```diff
@@ -1267,106 +1334,171 @@ This message consists of:
       specify the batch window instead.]
 
   The indicated query type MUST match the task's query type. Otherwise, the
-  Helper MUST abort with error "invalidMessage".
+  Helper MUST abort with error `invalidMessage`.
 
   This field is called the "partial" batch selector because, depending on the
   query type, it may not determine a batch. In particular, if the query type is
   `time_interval`, the batch is not determined until the Collector's query is
   issued (see {{query}}).
 
-* `report_shares`: The sequence of report shares to aggregate. The
-  `encrypted_input_share` field of the report share is the `HpkeCiphertext`
-  whose index in `Report.encrypted_input_shares` is equal to the index of the
-  aggregator in the task's `aggregator_endpoints` to which the
-  `AggregationJobInitReq` is being sent.
+* `prepare_steps`: the sequence of `PrepareStep` messages constructed in the
+  previous step.
 
-Let `{helper}` denote the Helper's API endpoint. The Leader sends a PUT
-request to `{helper}/tasks/{task-id}/aggregation_jobs/{aggregation-job-id}`
-with its `AggregationJobInitReq` message as the payload. The media type is
-"application/dap-aggregation-job-init-req". The Leader's aggregation job is now
-in round 0.
+Finally, the Leader sends a PUT request to
+`{helper}/tasks/{task-id}/aggregation_jobs/{aggregation-job-id}`, where
+`{helper}` denotes the Helper's API endpoint. The payload is set to
+`AggregationJobInitReq` and the media type is set to
+"application/dap-aggregation-job-init-req".
 
 The Helper's response will be an `AggregationJobResp` message (see
 {{aggregation-helper-init}}, which the Leader validates according to the
-criteria in {{aggregation-job-validation}}. If the message is valid, the Leader
-moves to the aggregation job continuation phase with the enclosed prepare steps,
-as described in {{agg-continue-flow}}. Otherwise, the Leader should abandon the
-aggregation job entirely.
+criteria in {{aggregation-job-validation}}. If the message is invalid, the
+Leader MUST abort the aggregation job.
+
+Otherwise, the Leader proceeds as follows with each report:
+
+1. If the inbound prep step has type "continue", then the Leader computes
+
+   ~~~
+   (state, outbound) = VDAF.ping_pong_req(agg_param, state, inbound)
+   ~~~
+
+   where `inbound` is the message payload. If `state == Rejected()`, then the
+   Leader rejects the report and removes it from the candidate set. If `state ==
```

Comment:
I believe this is incorrect: If `state == Finished(out_share)` **but** `outbound != None`, then we need to send one more continuation request.